davidia/server/plotserver.py

```diff
@@ -226,9 +421,50 @@ async def handle_client(server: PlotServer, plot_id: str, socket: WebSocket):
                     omit = client  # omit originating client
 
                 # currently used to test websocket communication in test_api
-                server.prepare_data(received_message, omit_client=omit)
+                await server.prepare_data(received_message, omit_client=omit)
                 await server.send_next_message()
 
     except WebSocketDisconnect:
         logger.error("Websocket disconnected:", exc_info=True)
         server.remove_client(plot_id, client)
+
+
+def add_indices(msg: MultiLineDataMessage) -> MultiLineDataMessage:
+    """Adds default indices to a multi-line data message if default indices flag is True
+
+    Parameters
+    ----------
+    msg : MultiLineDataMessage
+        A multi-line data message to which to add indices.
+
+    Returns the new multi-line data message
+    """
+    if msg.ml_data[0].default_indices:
+        for i, m in enumerate(msg.ml_data):
+            msg.ml_data[i].x = np.arange(m.y.size, dtype=m.y.dtype)
```

Comment:
The assignment can be to the `LineData` object `m` so then you don't need an `enumerate`